README.md
# NAME

Dist::Zilla::Plugin::Stenciller::HtmlExamples - Create Html example files from text files parsed with Stenciller

<div>
    <p>
    <img src="https://img.shields.io/badge/perl-5.10+-blue.svg" alt="Requires Perl 5.10+" />
    <a href="https://travis-ci.org/Csson/p5-Dist-Zilla-Plugin-Stenciller-HtmlExamples"><img src="https://api.travis-ci.org/Csson/p5-Dist-Zilla-Plugin-Stenciller-HtmlExamples.svg?branch=master" alt="Travis status" /></a>
    <a href="http://cpants.cpanauthors.org/dist/Dist-Zilla-Plugin-Stenciller-HtmlExamples-0.0201"><img src="https://badgedepot.code301.com/badge/kwalitee/Dist-Zilla-Plugin-Stenciller-HtmlExamples/0.0201" alt="Distribution kwalitee" /></a>
    <a href="http://matrix.cpantesters.org/?dist=Dist-Zilla-Plugin-Stenciller-HtmlExamples%200.0201"><img src="https://badgedepot.code301.com/badge/cpantesters/Dist-Zilla-Plugin-Stenciller-HtmlExamples/0.0201" alt="CPAN Testers result" /></a>
    <img src="https://img.shields.io/badge/coverage-98.5%-yellow.svg" alt="coverage 98.5%" />
    </p>
</div>

# VERSION

Version 0.0201, released 2016-03-23.

# SYNOPSIS

    ; in dist.ini
    ; these are the defaults
    [Stenciller::HtmlExamples]
    source_directory = examples/source
    output_directory = examples
    template_file = examples/source/template.html
    file_pattern = .+\.stencil
    output_also_as_html = 0

# DESCRIPTION

Dist::Zilla::Plugin::Stenciller::HtmlExamples uses [Stenciller](https://metacpan.org/pod/Stenciller) and [Stenciller::Plugin::ToHtmlPreBlock](https://metacpan.org/pod/Stenciller::Plugin::ToHtmlPreBlock) to turn
stencil files in `source_directory` (that matches the `file_pattern`) into
html example files in `output_directory` by applying the `template_file`.

This [Dist::Zilla](https://metacpan.org/pod/Dist::Zilla) plugin does the `FileGatherer` role.

# ATTRIBUTES

## source\_directory

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Path::Tiny#Dir">Dir</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>examples/source</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>Path to where the stencil files are.</p>

## output\_directory

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Path::Tiny#Path">Path</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>examples</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>Path to where the generated files are saved.  The output files
will have the same basename as the stencil they are based on, but with the suffix replaced by C<html>.</p>

## file\_pattern

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Str">Str</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>.+\.stencil</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>This is used as a part of a regular expression (so do not use start and end anchors) to find stencil files in the C<source_directory>.</p>

## template\_file

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Path::Tiny#AbsFile">AbsFile</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>&#39;template.html&#39; in <a href="#source_directory">&quot;source_directory&quot;</a></code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>The template file should be an html file. The first occurence of C<[STENCILS]> will be replaced with the output from L<Stenciller::Plugin::ToHtmlPreBlock>
for each stencil.</p>

## separator

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Maybe">Maybe</a> [ <a href="https://metacpan.org/pod/Types::Standard#Str">Str</a> ]</td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional</td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>Passed on to the L<Stenciller::Plugin::ToHtmlPreBlock> constructor.</p>

## output\_also\_as\_html

<table cellpadding="0" cellspacing="0">
<tr>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;"><a href="https://metacpan.org/pod/Types::Standard#Bool">Bool</a></td>
    <td style="padding-right: 6px; padding-left: 6px; border-right: 1px solid #b8b8b8; white-space: nowrap;">optional, default: <code>0</code></td>
    <td style="padding-left: 6px; padding-right: 6px; white-space: nowrap;">read-only</td>
</tr>
</table>

<p>Passed on to the L<Stenciller::Plugin::ToHtmlPreBlock> constructor.</p>

# SEE ALSO

- [Stenciller](https://metacpan.org/pod/Stenciller)
- [Stenciller::Plugin::ToHtmlPreBlock](https://metacpan.org/pod/Stenciller::Plugin::ToHtmlPreBlock)

# SOURCE

[https://github.com/Csson/p5-Dist-Zilla-Plugin-Stenciller-HtmlExamples](https://github.com/Csson/p5-Dist-Zilla-Plugin-Stenciller-HtmlExamples)

# HOMEPAGE

[https://metacpan.org/release/Dist-Zilla-Plugin-Stenciller-HtmlExamples](https://metacpan.org/release/Dist-Zilla-Plugin-Stenciller-HtmlExamples)

# AUTHOR

Erik Carlsson <info@code301.com>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Erik Carlsson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Dist-Zilla-Plugin-Stenciller-HtmlExamples

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.0201 2016-03-23T20:10:36Z UTC
 [Bug Fixes]
 - Fix compatibility with latest Stenciller (0.1400) (fixes #2).

 [Requirements]
 - Perl requirement lowered to 5.10 (from 5.14).

 [Stats & Dependency Changes]
 - 2 files changed, 13 insertions(+), 6 deletions(-)
 - (run req) + Types::Path::Tiny 0.005
 - (run req) + Types::Standard 1.000005
 - (run req) ~ Moose 2.1400 --> 2.1405
 - (run req) ~ MooseX::AttributeDocumented (any) --> 0.1003
 - (run req) ~ Path::Tiny (any) --> 0.072
 - (run req) ~ Stenciller 0.1302 --> 0.1400
 - (run req) ~ namespace::autoclean 0.18 --> 0.22
 - (run req) ~ perl 5.014000 --> 5.010000
 - (test req) ~ Path::Tiny (any) --> 0.072
 - (dev req) + Badge::Depot 0.0103
 - (dev req) + Badge::Depot::Plugin::Coverage (any)
 - (dev req) + Badge::Depot::Plugin::Cpantesters (any)
 - (dev req) + Badge::Depot::Plugin::Kwalitee (any)
 - (dev req) + Badge::Depot::Plugin::Perl (any)
 - (dev req) + Badge::Depot::Plugin::Travis (any)
 - (dev req) + Dist::Zilla 5.000
 - (dev req) + Dist::Zilla::Plugin::Authority 1.009
 - (dev req) + Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.007
 - (dev req) + Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0100
 - (dev req) + Dist::Zilla::Plugin::Git 2.036
 - (dev req) + Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.013
 - (dev req) + Dist::Zilla::Plugin::Git::Commit (any)
 - (dev req) + Dist::Zilla::Plugin::NextRelease::Grouped 0.0200
 - (dev req) + Dist::Zilla::Plugin::RewriteVersion::Transitional 0.007
 - (dev req) + Pod::Weaver::Section::GenerateSection 1.01
 - (dev req) + Test::EOF (any)
 - (dev req) + Test::Version 1
 - (dev req) ~ Dist::Zilla::Plugin::CheckChangesHasContent (any) --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::Clean (any) --> 0.07
 - (dev req) ~ Dist::Zilla::Plugin::CopyFilesFromBuild (any) --> 0.151680
 - (dev req) ~ Dist::Zilla::Plugin::Git::Contributors (any) --> 0.017
 - (dev req) ~ Dist::Zilla::Plugin::GithubMeta (any) --> 0.54
 - (dev req) ~ Dist::Zilla::Plugin::InstallRelease (any) --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::MetaProvides::Package (any) -->
   2.003001
 - (dev req) ~ Dist::Zilla::Plugin::PodWeaver (any) --> 4.006
 - (dev req) ~ Dist::Zilla::Plugin::ReadmeAnyFromPod (any) --> 0.150250
 - (dev req) ~ Dist::Zilla::Plugin::RunExtraTests (any) --> 0.028
 - (dev req) ~ Dist::Zilla::Plugin::Test::Compile (any) --> 2.054
 - (dev req) ~ Dist::Zilla::Plugin::Test::EOF (any) --> 0.0501
 - (dev req) ~ Dist::Zilla::Plugin::Test::EOL (any) --> 0.18
 - (dev req) ~ Dist::Zilla::Plugin::Test::Kwalitee::Extra (any) --> v0.2.1
 - (dev req) ~ Dist::Zilla::Plugin::Test::NoTabs (any) --> 0.15
 - (dev req) ~ Dist::Zilla::Plugin::Test::ReportPrereqs (any) --> 0.021
 - (dev req) ~ Dist::Zilla::Plugin::Test::Version (any) --> 1.05
 - (dev req) ~ Dist::Zilla::Plugin::TravisYML (any) --> 1.15
 - (dev req) ~ Pod::Elemental::Transformer::Splint (any) --> 0.1201
 - (dev req) ~ Pod::Weaver::Section::Badges (any) --> 0.0401
 - (dev req) ~ Test::More (any) --> 0.88
 - (dev req) ~ Test::NoTabs (any) --> 1.4
 - (dev req) ~ Test::Warnings (any) --> 0.026
 - (dev req) - Dist::Iller::Config::Author::CSSON
 - (dev req) - Dist::Zilla::Plugin::MetaProvides::Class
 - (dev req) - Dist::Zilla::Plugin::NextRelease
 - (dev req) - Dist::Zilla::Plugin::NextVersion::Semantic
 - (dev req) - Dist::Zilla::Plugin::OurPkgVersion
 - (dev req) - Dist::Zilla::Plugin::PodnameFromClassname
 - (dev req) - Dist::Zilla::Plugin::Prereqs::Plugins
 - (dev req) - Dist::Zilla::Plugin::PreviousVersion::Changelog
 - (dev req) - Dist::Zilla::Plugin::ReversionOnRelease
 - (dev req) - Pod::Weaver::Section::Homepage::DefaultCPAN
 - (dev req) - Pod::Weaver::Section::Source::DefaultGitHub
 - (dev sug) + Dist::Iller 0.1408
 - (dev sug) + Dist::Iller::Config::Author::CSSON 0.0309

0.0200 2016-01-27T18:29:46Z UTC
 [Other Changes]
 - No longer uses block package syntax, so should work on 5.10+ (once
   Stenciller does).

 [Requirements]
 - New prereq: namespace::autoclean.
 - Added missing prereqs: Moose, Path::Tiny.

0.0102 2016-01-10T15:34:21Z UTC
 [Bug Fixes]
 - Fixed failing test when used with latest Stenciller version (fixes #1).

 [Documentation]
 - Improved documentation.

 [Other Changes]
 - Regenerated dist.ini and weaver.ini from latest
   Dist::Iller::Config::Author::CSSON.

 [Requirements]
 - Bump Stenciller requirement to 0.1302, due to a change in
   Stenciller::Plugin::ToHtmlPreBlock output.
 - New requirements: MooseX::AttributeDocumented, String::Stomp and
   Syntax::Feature::Qs.

0.0101 2015-11-23T15:27:02Z UTC
 [Enhancements]
 - Added support for new attributes in Stenciller::Plugin::ToHtmlPreBlock.

 [Requirements]
 - Bump Stenciller requirement to 0.1301.

0.0100 2015-11-23T00:33:45Z UTC
 - Initial release


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.043.
use strict;
use warnings;

use 5.010000;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Create Html example files from text files parsed with Stenciller",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Stenciller-HtmlExamples",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010000",
  "NAME" => "Dist::Zilla::Plugin::Stenciller::HtmlExamples",
  "PREREQ_PM" => {
    "Dist::Zilla::File::InMemory" => 0,
    "Moose" => "2.1405",
    "MooseX::AttributeDocumented" => "0.1003",
    "Path::Tiny" => "0.072",
    "Stenciller" => "0.1400",
    "String::Stomp" => 0,
    "Syntax::Feature::Qs" => "0.2004",
    "Types::Path::Tiny" => "0.005",
    "Types::Standard" => "1.000005",
    "namespace::autoclean" => "0.22",
    "syntax" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Path::Tiny" => "0.072",
    "String::Cushion" => 0,
    "Syntax::Feature::Qi" => 0,
    "Test::DZil" => 0,
    "Test::Differences" => 0,
    "Test::More" => "0.96",
    "syntax" => 0
  },
  "VERSION" => "0.0202",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::File::InMemory" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => "2.1405",
  "MooseX::AttributeDocumented" => "0.1003",
  "Path::Tiny" => "0.072",
  "Stenciller" => "0.1400",
  "String::Cushion" => 0,
  "String::Stomp" => 0,
  "Syntax::Feature::Qi" => 0,
  "Syntax::Feature::Qs" => "0.2004",
  "Test::DZil" => 0,
  "Test::Differences" => 0,
  "Test::More" => "0.96",
  "Types::Path::Tiny" => "0.005",
  "Types::Standard" => "1.000005",
  "namespace::autoclean" => "0.22",
  "syntax" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
